langkit/templates/pkg_analysis_spec_ada.mako

```diff
@@ -109,6 +109,63 @@ package ${ada_lib_name}.Analysis is
    --  Return a short string describing ``Node``, or None" if ``Node.Is_Null``
    --  is true.
 
+   -------------------
+   -- Event handler --
+   -------------------
+
+   type Event_Handler_Interface is interface;
+   ${ada_doc('langkit.event_handler_type')}
+
+   procedure Unit_Requested_Callback
+     (Self               : Event_Handler_Interface;
+      Context            : Analysis_Context'Class;
+      Name               : Text_Type;
+      From               : Analysis_Unit'Class;
+      Found              : Boolean;
+      Is_Not_Found_Error : Boolean) is null;
+   --  Callback that will be called when an unit is requested from the context
+   --  ``Context``.
+   --
+   --  ``Name`` is the name of the requested unit.
+   --
+   --  ``From`` is the unit from which the unit was requested.
+   --
+   --  ``Found`` indicates whether the requested unit was found or not.
+   --
+   --  ``Is_Not_Found_Error`` indicates whether the fact that the unit was not
+   --  found is an error or not.
```

Comment:
Sorry, I should have made this explicit in my previous comment: what about using the `ada_doc` helper to avoid duplication with docs from `langkit.documentation`?

Reply:
It's not possible because only callbacks types are documented in langkit.documentation, not callback functions. You did the exact same thing for the file reader btw.

Reply:
I did that for file readers because the file reader interface is defined in Langkit_Support. Here, everything is in Mako templates, so this should be possible, no?

Reply:
Except that the first part of the comment is not exactly the same: one refers to a callback, the other to a callback type

Reply:
Ah yes, you are right! My bad, all is good, thanks.

(we should probably eventually find a way to share the common part, but that’s out of the scope of this change)